src/measure_theory/function/continuous_map_dense.lean

```diff
@@ -42,152 +58,332 @@ Vitali-Carathéodory theorem, in the file `measure_theory.vitali_caratheodory`.
+  have h'ε : ε / 2 ≠ 0, by simpa using hε,
+  rcases exists_snorm_indicator_le hp c h'ε with ⟨η, ηpos, hη⟩,
+  have hη_pos' : (0 : ℝ≥0∞) < η := ennreal.coe_pos.2 ηpos,
+  obtain ⟨s, st, s_closed, μs⟩ : ∃ s ⊆ t, is_closed s ∧ μ (t \ s) < η :=
+    ht.exists_is_closed_diff_lt htμ.ne hη_pos'.ne',
+  have hsμ : μ s < ∞ := (measure_mono st).trans_lt htμ,
+  have I1 : snorm (s.indicator (λ y, c) - t.indicator (λ y, c)) p μ ≤ ε/2,
+  { rw [← snorm_neg, neg_sub, ← indicator_diff st],
+    exact (hη _ μs.le) },
+  rcases exists_continuous_snorm_sub_le_of_closed hp s_closed is_open_univ (subset_univ _)
+    hsμ.ne c h'ε with ⟨f, I2, f_cont, f_bound, -, f_mem⟩,
+  have I3 : snorm (f - t.indicator (λ y, c)) p μ ≤ ε, from calc
+    snorm (f - t.indicator (λ y, c)) p μ
+      = snorm ((f - s.indicator (λ y, c)) + (s.indicator (λ y, c) - t.indicator (λ y, c))) p μ :
+    by simp only [sub_add_sub_cancel]
+  ... ≤ snorm (f - s.indicator (λ y, c)) p μ
+        + snorm (s.indicator (λ y, c) - t.indicator (λ y, c)) p μ :
+    begin
+      apply snorm_add_le _ _ h'p,
+      { exact f_mem.ae_strongly_measurable.sub
+          (ae_strongly_measurable_const.indicator s_closed.measurable_set) },
+      { exact (ae_strongly_measurable_const.indicator s_closed.measurable_set).sub
+          (ae_strongly_measurable_const.indicator ht) },
+    end
+  ... ≤ ε/2 + ε/2 : add_le_add I2 I1
+  ... = ε : ennreal.add_halves _,
+  refine ⟨f, I3, f_cont, f_mem, _⟩,
+  exact (bounded_continuous_function.of_normed_add_comm_group f f_cont _ f_bound).bounded_range,
+end
+
+/-- Any function in `ℒp` can be approximated by bounded continuous functions when `1 ≤ p < ∞`,
+version in terms of `∫`. -/
+lemma mem_ℒp.exists_bounded_continuous_integral_rpow_sub_le
+  [μ.weakly_regular] {p : ℝ} (h'p : 1 ≤ p)
+  {f : α → E} (hf : mem_ℒp f (ennreal.of_real p) μ) {ε : ℝ} (hε : 0 < ε) :
+  ∃ (g : α →ᵇ E), ∫ x, ‖f x - g x‖^p ∂μ ≤ ε ∧ mem_ℒp g (ennreal.of_real p) μ :=
+begin
+  have I : 0 < ε ^ (1/p) := real.rpow_pos_of_pos hε _,
+  have A : ennreal.of_real (ε ^ (1/p)) ≠ 0,
+    by simp only [ne.def, ennreal.of_real_eq_zero, not_le, I],
+  have B : 1 ≤ ennreal.of_real p,
+  { convert ennreal.of_real_le_of_real h'p, exact ennreal.of_real_one.symm },
+  rcases hf.exists_bounded_continuous_snorm_sub_le ennreal.coe_ne_top B A
+    with ⟨g, hg, g_mem⟩,
+  change snorm _ (ennreal.of_real p) _ ≤ _ at hg,
+  refine ⟨g, _, g_mem⟩,
+  rwa [(hf.sub g_mem).snorm_eq_integral_rpow_norm (zero_lt_one.trans_le B).ne'
+    ennreal.coe_ne_top, ennreal.of_real_le_of_real_iff I.le, one_div,
+    ennreal.to_real_of_real (zero_le_one.trans h'p), real.rpow_le_rpow_iff _ hε.le _] at hg,
+  { exact integral_nonneg (λ x, real.rpow_nonneg_of_nonneg (norm_nonneg _) _) },
+  { exact inv_pos.2 (zero_lt_one.trans_le h'p) }
+end
+
+/-- Any integrable function can be approximated by bounded continuous functions,
+version in terms of `∫⁻`. -/
+lemma integrable.exists_bounded_continuous_lintegral_sub_le [μ.weakly_regular]
```

Comment:
Maybe not, but who knows. Often when dealing with norms, the ennreal-valued integral is nicer to use than the real one. Since it's really short, I'd rather keep it.